opencti-platform/opencti-graphql/src/modules/ingestion/ingestion-csv-domain.ts

```diff
@@ -43,18 +49,54 @@ export const findCsvMapperForIngestionById = (context: AuthContext, user: AuthUs
   return storeLoadById<BasicStoreEntityCsvMapper>(context, user, csvMapperId, ENTITY_TYPE_CSV_MAPPER);
 };
 
+export const createOnTheFlyUser = async (context: AuthContext, user: AuthUser, input: IngestionCsvAddInput) => {
+  const groups = await findAll(context, context.user, {
+    filters: {
+      mode: 'and',
+      filters: [
+        {
+          key: ['auto_integration_assignation'],
+          values: [
+            'global',
+          ],
+        },
+      ],
+      filterGroups: [],
+    }
+  });
+  if (groups.edges.length === 0) {
+    throw FunctionalError('You have not defined a default group for ingestion users', {});
+  }
+  const { platform_organization } = await getEntityFromCache<BasicStoreSettings>(context, SYSTEM_USER, ENTITY_TYPE_SETTINGS);
+  const newlyCreatedUser = await addUser(context, context.user, {
+    user_email: `${input.user_id}@opencti.invalid`,
+    name: input.user_id,
+    prevent_default_groups: true,
+    groups: groups.edges[0].node,
```

Comment:
maybe better to add connectionFormat: false in opts, and avoid having edges and nodes.